networkx/algorithms/assortativity/connectivity.py

```diff
@@ -17,7 +17,7 @@ def _avg_deg_conn(G, neighbors, source_degree, target_degree,
     dsum = defaultdict(float)
     dnorm = defaultdict(float)
     for n,k in source_degree(nodes).items():
-        nbrdeg = target_degree(neighbors(n))
+        nbrdeg = target_degree(list(neighbors(n)))
```

Comment:
It looks like D.degree is not handling an iterator correctly...???
Same thing happens on the Master branch with digraph.degree(). 
I think there is a bug in digraph.degree_iter()
Line 928
It calls nbunch_iter twice with the same iterator.

:)
Wonder how long THAT bug has been there. :)
I think that should probably be fixed in 1.10 as well as here.